internal/mage/sdk/python.go

```diff
@@ -90,7 +90,39 @@ func (t Python) Generate(ctx context.Context) error {
 
 // Publish publishes the Python SDK
 func (t Python) Publish(ctx context.Context, tag string) error {
-	return errors.New("not implemented")
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	var (
+		version = strings.TrimPrefix(tag, "sdk/python/v")
+		token   = os.Getenv("PYPI_TOKEN")
+		// "test" for test.pypi.org, empty for public one
+		repo = os.Getenv("PYPI_REPO")
+	)
+
+	if token == "" {
+		return errors.New("PYPI_TOKEN environment variable must be set")
+	}
+
+	build := pythonBase(c).
```

Comment:
With #3724 you'll have to wrap this in a `util.WithDevEngine`

See https://github.com/dagger/dagger/pull/3724/files#diff-a81a610a4cd1cce31473f5175bc1df43001f3a1d85d010577195aa52d5025ac8R112-R118

Reply:
Publish doesn't need the engine, is it still needed?

Reply:
Oh, I see... for running the Go SDK stuff.

Reply:
@aluzzardi When is `util.WithDevEngine` needed, really? Isn't it when the python code needs to connect to the API?

Reply:
Based on a sync conversation, I also don't think that we need `WithDevEngine` (a.k.a. "dagger in dagger") since the command running inside the container is `poetry` which doesn't need access to Dagger.